### ISSUE TITLE
IPNs from blocknonomics.co are blocked

### ISSUE THREAD

**Issue by CherryDT** [localtunnel/localtunnel#376]
I know you have this disclaimer page coming up once in a while, but it says IPNs are not blocked.

I had an issue yesterday though, while I was developing another payment integration on a project I'm working on: Blockonomics. I didn't seem to get any IPNs. I then checked in the IPN log on their side, and it showed the IPNs sent and successfully received with a HTTP 200 response, yet my service never saw them (let alone responded), so I can only conclude that it's your disclaimer page interfering.

I used ngrok for this test then, but maybe you want to take a look at that.

**Comment by CherryDT** [comment]
Hm, I did another test now with a requestbin to see what is sent exactly: http://requestbin.net/r/13auurh1?inspect

I guess the problem is that they are sending a browser-like user agent. I'm not sure what to do about it then, other than adding exceptions based on specific parameters existing in requests, or something like that. How do you solve that with other services which do that? Or is that the first one you encounter?

Maybe the info message should be updated to warn that if the server sends a browser-like user agent, it will be blocked anyway.

**Comment by TheBoroer** [comment]
Thanks for the feedback! 
I've just updated the tunnel reminder page to kind of explain it more along with outlining 2 options on how to bypass the tunnel reminder page ([https://example.loca.lt](https://example.loca.lt)).

To be honest this is the first service who's IPN has a browser-linke user agent set. Can you get them to set an additional custom request header? I also changed the reminder page to return an HTTP 401 response code

**Comment by CherryDT** [comment]
Look like I can't get them to send a header.
Anyway, thanks for the update!

**Comment by AJ-54** [comment]
Hi @CherryDT, I tried to test the blockonomics payment API on my local using localtunnel and it is working fine for me. I receive the response on the HTTP callback URL that I place.

**Comment by CherryDT** [comment]
Weird, maybe they changed the headers? You can see the request I got at http://requestbin.net/r/13auurh1?inspect (and it caused a 200 response which didn't come from my local server)

**Comment by blockonomics** [comment]
Blockonomics reprsentative here. We found this from search results. Yes in fact we set a browser *User Agent*. The reason behind that there are VPS servers used by e-commerce merchants that routine reject requests from from browser agents. 

Is there a way to configure localtunnel to allow this particular UserAgent?

**Comment by TheBoroer** [comment]
Hey @blockonomics
Great to have you chime in!
You mean there are VPS servers that reject requests from NON-browser agents, so that's why you use standard browser user agents.

Are you able to add a blocknomics specific http request header to all your IPN requests (e.g. x-blocknomics) that I can use to distiguish your IPN request from regular browser user requests?

**Comment by blockonomics** [comment]
> Are you able to add a blocknomics specific http request header to all your IPN requests (e.g. X-BLOCKNOMICS or BLOCKNOMICS-API-VERSION etc) that I can use to distiguish your IPN request from regular browser user requests?

Yes this is possible. What should be recommend value of such a header?

**Comment by blockonomics** [comment]
We have now removed the Browser User Agent
@CherryDT  please check now. Let us know if you still face any problems

**Comment by blockonomics** [comment]
Removing Browser User agent has caused the following issue https://stackoverflow.com/questions/23190351/request-works-fine-in-browser-but-403-in-python/23200605#23200605

So we have reverted this and added a new header
`'X-BLOCKONOMICS-VERSION': '1'`

**Comment by TheBoroer** [comment]
Perfect! Thanks for the update @blockonomics 
I've added the http header `X-BLOCKONOMICS-VERSION` to bypass our tunnel reminder page.
